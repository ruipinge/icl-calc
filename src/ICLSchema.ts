import * as Yup from 'yup';

const INVALID_ERROR = 'Invalid value.';
const REQUIRED_ERROR = 'Required value.';

export const ICLSchema = Yup.object().shape({
  patient: Yup.object().shape({
    name: Yup.string()
      .min(5, 'Too Short!')
      .max(10, 'Too Long!')
      .required('Required')
  }),
  biometrics: Yup.object().shape({
    ata: Yup.number()
      .required(REQUIRED_ERROR)
      .min(0, INVALID_ERROR)
      .max(50, INVALID_ERROR),
    wtw: Yup.number()
      .required(REQUIRED_ERROR)
      .min(-50, INVALID_ERROR)
      .max(50, INVALID_ERROR),
    clr: Yup.number()
      .required(REQUIRED_ERROR)
      .min(-1000, INVALID_ERROR)
      .max(1000, INVALID_ERROR),
    acd: Yup.number()
      .required(REQUIRED_ERROR)
      .min(-50, INVALID_ERROR)
      .max(50, INVALID_ERROR),
    acan: Yup.number()
      .required(REQUIRED_ERROR)
      .min(0, INVALID_ERROR)
      .max(180, INVALID_ERROR),
    acat: Yup.number()
      .required(REQUIRED_ERROR)
      .min(0, INVALID_ERROR)
      .max(180, INVALID_ERROR),
    kfm: Yup.number()
      .required(REQUIRED_ERROR)
      .min(-1000, INVALID_ERROR)
      .max(1000, INVALID_ERROR),
    cct: Yup.number()
      .required(REQUIRED_ERROR)
      .min(-1000, INVALID_ERROR)
      .max(1000, INVALID_ERROR)
  }),
  spectacleRefraction: Yup.object().shape({
    sphere: Yup.number()
      .required(REQUIRED_ERROR)
      .min(-1000, INVALID_ERROR)
      .max(1000, INVALID_ERROR),
    cylindre: Yup.number()
      .required(REQUIRED_ERROR)
      .min(-1000, INVALID_ERROR)
      .max(1000, INVALID_ERROR),
    axis: Yup.number()
      .required(REQUIRED_ERROR)
      .min(0, INVALID_ERROR)
      .max(180, INVALID_ERROR)
  })
});
