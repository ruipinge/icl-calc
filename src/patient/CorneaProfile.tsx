import { Field, FormikState } from 'formik';
import { ICLInputs, PreviousSurgery } from '../types';

import { FieldWithUnit } from './FieldWithUnit';

export const CorneaProfile = ({
  errors,
  touched
}: FormikState<Pick<ICLInputs, 'corneaProfile'>>) => (
  <div className="section-form">
    <h4>Cornea Profile</h4>
    <FieldWithUnit
      label="Anterior Keratometry Flat (KAntFlt)"
      name="corneaProfile.kaf"
      unit="dpt"
      error={errors.corneaProfile?.kaf}
      touched={touched.corneaProfile?.kaf}
    />
    <FieldWithUnit
      label="@"
      name="corneaProfile.axisaf"
      unit="º"
      error={errors.corneaProfile?.axisaf}
      touched={touched.corneaProfile?.axisaf}
    />
    <FieldWithUnit
      label="Anterior Keratometry Steep (KAntStp)"
      name="corneaProfile.kas"
      unit="dpt"
      error={errors.corneaProfile?.kas}
      touched={touched.corneaProfile?.kas}
    />
    <FieldWithUnit
      label="@"
      name="corneaProfile.axisas"
      unit="º"
      error={errors.corneaProfile?.axisas}
      touched={touched.corneaProfile?.axisas}
    />
    <FieldWithUnit
      label="Posterior Keratometry Flat (KPostFlt)"
      name="corneaProfile.kpf"
      unit="dpt"
      error={errors.corneaProfile?.kpf}
      touched={touched.corneaProfile?.kpf}
    />
    <FieldWithUnit
      label="@"
      name="corneaProfile.axispf"
      unit="º"
      error={errors.corneaProfile?.axispf}
      touched={touched.corneaProfile?.axispf}
    />
    <FieldWithUnit
      label="Posterior Keratometry Steep (KPostStp)"
      name="corneaProfile.kps"
      unit="dpt"
      error={errors.corneaProfile?.kps}
      touched={touched.corneaProfile?.kps}
    />
    <FieldWithUnit
      label="@"
      name="corneaProfile.axisps"
      unit="º"
      error={errors.corneaProfile?.axisps}
      touched={touched.corneaProfile?.axisps}
    />
    <FieldWithUnit
      label="Central Corneal Thickness (CCT)"
      name="corneaProfile.cct"
      unit="μm"
      error={errors.corneaProfile?.cct}
      touched={touched.corneaProfile?.cct}
    />
    <div className={'form-group row mb-4'}>
      <label
        htmlFor={'corneaProfile.previousSurgery'}
        className="col-sm-6 col-form-label"
      >
        Previous Corneal Refractive Surgery
      </label>
      <div className="col-sm-6">
        <div className="input-group">
          <Field
            as="select"
            name="corneaProfile.previousSurgery"
            className="form-control"
          >
            {Object.values(PreviousSurgery).map((val) => (
              <option value={val} key={val}>
                {val}
              </option>
            ))}
          </Field>
        </div>
      </div>
    </div>
  </div>
);
