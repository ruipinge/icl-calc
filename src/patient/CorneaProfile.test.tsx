import { CorneaProfile } from './CorneaProfile';
import { Formik } from 'formik';
import { ICLSchema } from '../ICLSchema';
import TestRenderer from 'react-test-renderer';

it('renders without crashing', () => {
  const tree = TestRenderer.create(
    <Formik
      initialValues={{
        corneaProfile: {
          kf: 1,
          ks: 2,
          axisf: 3,
          axiss: 4,
          cct: 5
        }
      }}
      validationSchema={ICLSchema}
      onSubmit={() => {}}
    >
      {({ errors, touched, values, resetForm, ...otherProps }) => (
        <CorneaProfile
          errors={{}}
          values={values}
          touched={{}}
          {...otherProps}
        />
      )}
    </Formik>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
