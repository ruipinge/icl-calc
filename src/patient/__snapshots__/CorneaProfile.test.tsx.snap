// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="section-form"
>
  <h4>
    Cornea Profile
  </h4>
  <div
    className="form-group row"
  >
    <label
      className="col-sm-6 col-form-label"
      htmlFor="corneaProfile.kaffield"
    >
      Anterior Keratometry Flat (KAntFlt)
    </label>
    <div
      className="col-sm-6"
    >
      <div
        className="input-group"
      >
        <input
          autoComplete="off"
          className="form-control text-right"
          name="corneaProfile.kaf"
          onBlur={[Function]}
          onChange={[Function]}
          type="number"
        />
        <div
          className="input-group-append"
        >
          <span
            className="input-group-text"
            title="dioptres"
          >
            dpt
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="form-group row"
  >
    <label
      className="col-sm-6 col-form-label"
      htmlFor="corneaProfile.axisaffield"
    >
      @
    </label>
    <div
      className="col-sm-6"
    >
      <div
        className="input-group"
      >
        <input
          autoComplete="off"
          className="form-control text-right"
          name="corneaProfile.axisaf"
          onBlur={[Function]}
          onChange={[Function]}
          type="number"
        />
        <div
          className="input-group-append"
        >
          <span
            className="input-group-text"
            title="degrees"
          >
            º
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="form-group row"
  >
    <label
      className="col-sm-6 col-form-label"
      htmlFor="corneaProfile.kasfield"
    >
      Anterior Keratometry Steep (KAntStp)
    </label>
    <div
      className="col-sm-6"
    >
      <div
        className="input-group"
      >
        <input
          autoComplete="off"
          className="form-control text-right"
          name="corneaProfile.kas"
          onBlur={[Function]}
          onChange={[Function]}
          type="number"
        />
        <div
          className="input-group-append"
        >
          <span
            className="input-group-text"
            title="dioptres"
          >
            dpt
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="form-group row"
  >
    <label
      className="col-sm-6 col-form-label"
      htmlFor="corneaProfile.axisasfield"
    >
      @
    </label>
    <div
      className="col-sm-6"
    >
      <div
        className="input-group"
      >
        <input
          autoComplete="off"
          className="form-control text-right"
          name="corneaProfile.axisas"
          onBlur={[Function]}
          onChange={[Function]}
          type="number"
        />
        <div
          className="input-group-append"
        >
          <span
            className="input-group-text"
            title="degrees"
          >
            º
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="form-group row"
  >
    <label
      className="col-sm-6 col-form-label"
      htmlFor="corneaProfile.kpffield"
    >
      Posterior Keratometry Flat (KPostFlt)
    </label>
    <div
      className="col-sm-6"
    >
      <div
        className="input-group"
      >
        <input
          autoComplete="off"
          className="form-control text-right"
          name="corneaProfile.kpf"
          onBlur={[Function]}
          onChange={[Function]}
          type="number"
        />
        <div
          className="input-group-append"
        >
          <span
            className="input-group-text"
            title="dioptres"
          >
            dpt
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="form-group row"
  >
    <label
      className="col-sm-6 col-form-label"
      htmlFor="corneaProfile.axispffield"
    >
      @
    </label>
    <div
      className="col-sm-6"
    >
      <div
        className="input-group"
      >
        <input
          autoComplete="off"
          className="form-control text-right"
          name="corneaProfile.axispf"
          onBlur={[Function]}
          onChange={[Function]}
          type="number"
        />
        <div
          className="input-group-append"
        >
          <span
            className="input-group-text"
            title="degrees"
          >
            º
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="form-group row"
  >
    <label
      className="col-sm-6 col-form-label"
      htmlFor="corneaProfile.kpsfield"
    >
      Posterior Keratometry Steep (KPostStp)
    </label>
    <div
      className="col-sm-6"
    >
      <div
        className="input-group"
      >
        <input
          autoComplete="off"
          className="form-control text-right"
          name="corneaProfile.kps"
          onBlur={[Function]}
          onChange={[Function]}
          type="number"
        />
        <div
          className="input-group-append"
        >
          <span
            className="input-group-text"
            title="dioptres"
          >
            dpt
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="form-group row"
  >
    <label
      className="col-sm-6 col-form-label"
      htmlFor="corneaProfile.axispsfield"
    >
      @
    </label>
    <div
      className="col-sm-6"
    >
      <div
        className="input-group"
      >
        <input
          autoComplete="off"
          className="form-control text-right"
          name="corneaProfile.axisps"
          onBlur={[Function]}
          onChange={[Function]}
          type="number"
        />
        <div
          className="input-group-append"
        >
          <span
            className="input-group-text"
            title="degrees"
          >
            º
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="form-group row"
  >
    <label
      className="col-sm-6 col-form-label"
      htmlFor="corneaProfile.cctfield"
    >
      Central Corneal Thickness (CCT)
    </label>
    <div
      className="col-sm-6"
    >
      <div
        className="input-group"
      >
        <input
          autoComplete="off"
          className="form-control text-right"
          name="corneaProfile.cct"
          onBlur={[Function]}
          onChange={[Function]}
          type="number"
          value={5}
        />
        <div
          className="input-group-append"
        >
          <span
            className="input-group-text"
            title="micrometres"
          >
            μm
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="form-group row mb-4"
  >
    <label
      className="col-sm-6 col-form-label"
      htmlFor="corneaProfile.previousSurgery"
    >
      Previous Corneal Refractive Surgery
    </label>
    <div
      className="col-sm-6"
    >
      <div
        className="input-group"
      >
        <select
          className="form-control"
          name="corneaProfile.previousSurgery"
          onBlur={[Function]}
          onChange={[Function]}
        >
          <option
            value="None"
          >
            None
          </option>
          <option
            value="Myopia"
          >
            Myopia
          </option>
          <option
            value="Hyperopia"
          >
            Hyperopia
          </option>
        </select>
      </div>
    </div>
  </div>
</div>
`;
